### PR TITLE
Updating LOIO analysis to include all features spaces

### PR DESCRIPTION
**Pull request by gwaybio**
I also add shuffled data to this analysis. Dependent on merging #40 first. After merging #40, I will update visualization

### CODE REVIEW

File: 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb [hunk 1]
```diff
@@ -0,0 +1,1258 @@
+{
```
Comment (MattsonCam) [inline 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: <p>May also consider removing this</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/phenotypic_profiling_model/pull/46/discussion/#comment-1376604959'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (MattsonCam) [inline 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: <p>Very well documented, easy to follow</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/phenotypic_profiling_model/pull/46/discussion/#comment-1376604962'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (MattsonCam) [inline 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #60.            straified_k_folds = StratifiedKFold(n_splits=10, shuffle=False)
</pre><p>Consider removing this code</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/phenotypic_profiling_model/pull/46/discussion/#comment-1376604963'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (MattsonCam) [inline 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #62.            # create logistic regression model with following parameters
</pre><p>Same thing here, consider removing this code</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/phenotypic_profiling_model/pull/46/discussion/#comment-1376604965'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (MattsonCam) [inline 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #68.            grid_search_cv = GridSearchCV(
</pre><p>Consider removing this code</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/phenotypic_profiling_model/pull/46/discussion/#comment-1376604969'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (MattsonCam) [inline 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #94.                        C=model.C,
</pre><p>Obviously it works, but I'm surprised you can access the C parameter, since it is not an attribute on the 1.1.1 sklearn logistic regression documentation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/phenotypic_profiling_model/pull/46/discussion/#comment-1376604975'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (MattsonCam) [inline 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    # define combinations to test over
</pre><p>May also consider storing these in a dictionary</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/phenotypic_profiling_model/pull/46/discussion/#comment-1376604978'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

**Review by MattsonCam — APPROVED** [review]
It LGTM @gwaybio! Just some minor comments. Let me know if you have any questions!

File: 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb (hunk 1 above)
Comment (gwaybio) [reply 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: Yeah, I agree. I will remove!

Comment (gwaybio) [reply 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: Great! It is maybe worth noting that this is mostly Roshan's code ported over to this new file.

Comment (gwaybio) [reply 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: Great point! this must have stuck around and I forgot to delete. Thanks for noting. (I will also delete the GridSearchCV object call)

Comment (gwaybio) [reply 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: 💯 thanks!

Comment (gwaybio) [reply 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: Interesting! Thanks for digging into this and raising. I confirmed that I am using 1.1.1 (see below), but maybe it is through the saving and/or loading with joblib process that exposes the attribute again? Maybe this process also exposes parameters (C is a parameter in [1.1 docs](https://scikit-learn.org/1.1/modules/generated/sklearn.linear_model.LogisticRegression.html?highlight=logisticregression#sklearn.linear_model.LogisticRegression)) 🤷 

image

Comment (gwaybio) [reply 3.evaluate_model/get_LOIO_probabilities_all_featurespaces_no_retraining.ipynb:1]: Yeah, it might be more elegant to do so. However, it is a relatively minor change that would cause additional compute time that likely isn't worth pursuing. Roshan also uses `itertools.product()` on these combinations, so it is efficiently iterating over them with limited code, which is one of the primary benefits of dictionary storage. I will skip doing this one, thanks!

**Comment by gwaybio** [comment]
Thanks for the review @MattsonCam - I will go ahead and merge!
